zigzap-porep.md

```diff
@@ -0,0 +1,281 @@
+# Filecoin ZigZag Proof of Replication
+
+ZigZagDrgPorep is a layered PoRep which replicates layer by layer.
+Between layers, the graph is 'reversed' in such a way that the dependencies expand with each iteration.
+This reversal is not a straightforward inversion -- so we coin the term 'zigzag' to describe the transformation.
+Each graph can be divided into base and expansion components.
+The 'base' component is an ordinary DRG. The expansion component attempts to add a target (expansion_degree) number of connections
+between nodes in a reversible way. Expansion connections are therefore simply inverted at each layer.
+Because of how DRG-sampled parents are calculated on demand, the base components are not. Instead, a same-degree
+DRG with connections in the opposite direction (and using the same random seed) is used when calculating parents on demand.
+For the algorithm to have the desired properties, it is important that the expansion components are directly inverted at each layer.
+However, it is fortunately not necessary that the base DRG components also have this property.
+
+## Hash Functions
+
+__*Filecoin ZigZag Proof of Replication*__ is the process by which raw data is transformed into a replica and a proof of replication.
+
+This comprises the following steps:
+ - Preprocessing
+ - Sector Padding
+ - Replication
+ - Proof Generation
+ - Circuit Proof Generation
+ 
+ Together, __*Replication*__, __*Proof Generation*__, and __*Circuit Proof Generation*__ constitute the `Seal` operation, as described in [Filecoin Proofs](proofs.md).
```

Comment:
up until this point this is general glossary, not hash functions; I would use a different heading.